mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -191,12 +191,35 @@ func (sp *StateProcessor[
 		math.Gwei(sp.cs.MaxEffectiveBalance()),
 	)
 
-	// TODO: This is a bug that lives on bArtio. Delete this eventually.
-	if sp.cs.DepositEth1ChainID() == bArtioChainID {
-		if err := st.AddValidatorBartio(val); err != nil {
+	if !sp.processingGenesis {
```

Comment:
minor optimization which we may drop for readability.
Currently I chose to fail node initialization if genesis has more validators than the validator set cap, so here I can skip the check.
However we may chose to allow genesis have more validators than cap and immediately mark the ready for eviction.
@itsdevbear @chuck-bear thoughts?